skylark/skylark.bzl

```diff
@@ -46,8 +30,10 @@ def _skydoc(ctx):
 def _skylark_doc_impl(ctx):
   """Implementation of the skylark_doc rule."""
   skylark_doc_zip = ctx.outputs.skylark_doc_zip
-  inputs = _get_transitive_sources(ctx.attr.deps) + ctx.files.srcs
-  sources = [source.path for source in inputs]
+  inputs = depset(order="postorder", transitive=[
+      dep[SkylarkLibraryInfo].transitive_srcs for dep in ctx.attr.deps
+  ]) + ctx.files.srcs
```

Comment:
do not use `+` operator for depsets, pass `ctx.files.srcs` as an element of `transitive`

Reply:
`transitive` expects a sequence of `depset`s, not `File`s. Placed it in the `direct` argument instead. I _think_ that does what you want.

Reply:
Done.